### DISCUSSION TITLE
Keeping two remotes in sync one LFS the other non-LFS.

### DISCUSSION THREAD

**Discussion by axkibe** [git-lfs/git-lfs#5139] [Q&A]
Hi again.

Following scenario: I got two remotes.. one our own git server which has LFS enabled (and is LFS object server at the same time) the other (sync to overleaf in this case) has no LFS.

Now I want to pull/push to the non-LFS server (overleaf remote), but I do want the binary objects to be there verbatim, not the LFS pointer files.

Do I need to constantly use migrate for and back (and I hope the histories keep lined up) or how do I approach that?

Or is there a pull/push overlay that migrates each commit to a classical verbatim one for that remote?

* (to be more exact this ought to be automatic code within my git server that pulls from overleaf before any git operation and pushes to overleaf remote after every receive-pack operation. I've done something before with a overleaf community edition using the webapi to up/download a zip file.. but now we're switching to the cloud edition and I'd like to use to git API since it's a much cleaner approach on the overleaf side)

**Comment by bk2204** [comment]
Hey,

Yes, in this case, you need to migrate back and forth every time.  The LFS and non-LFS histories will have completely different object IDs and divergent histories.  I wouldn't recommend that because it's a terrible experience and is going to cause untold problems, but if you want to do that, that is what's required.

**Comment by axkibe** [comment]
Thanks I experimented a little and I didn't manage to just migrate said new commits and got constant unrelated history errors.

I guess I will just keep the repositories apart git-wise and automatically sync them with rsync. whenever a push happens on either.
